src/ray/object_manager/object_directory.cc

```diff
@@ -86,25 +98,65 @@ ray::Status ObjectDirectory::GetInformation(const ClientID &client_id,
   return ray::Status::OK();
 }
 
-ray::Status ObjectDirectory::SubscribeObjectLocations(const ObjectID &object_id,
+ray::Status ObjectDirectory::SubscribeObjectLocations(const UniqueID &callback_id,
+                                                      const ObjectID &object_id,
                                                       const OnLocationsFound &callback) {
-  if (listeners_.find(object_id) != listeners_.end()) {
-    RAY_LOG(ERROR) << "Duplicate calls to SubscribeObjectLocations for " << object_id;
+  ray::Status status = ray::Status::OK();
+  if (listeners_.find(object_id) == listeners_.end()) {
+    listeners_.emplace(object_id, LocationListenerState());
+    status = gcs_client_->object_table().RequestNotifications(
+        JobID::nil(), object_id, gcs_client_->client_table().GetLocalClientId());
+  }
+  if (listeners_[object_id].callbacks.count(callback_id) > 0) {
```

Comment:
I think we should probably make this a fatal check, unless there is a valid case where this would happen.

Reply:
The only case I can think of is when a `Pull` is invoked multiple times for the same object. I think we should keep this the way it is until we implement `Pull` suppression (I've added a TODO). If we make this fatal now, we may cause unnecessary failures.